### ISSUE TITLE
Avoid using the discrete GPU on Macs

### ISSUE THREAD

**Issue by 3rror** [raspberrypi/rpi-imager#325]
Raspberry Pi Imager forces the use of the discrete GPU on the MacBook Pro 16" with Intel processor and AMD GPU, and probably on all Macs with two graphic units.

It's probably because it is missing the `NSSupportsAutomaticGraphicsSwitching` key in `Info.plist`.

```xml
[...]
<dict>
	[...]
	<key>NSSupportsAutomaticGraphicsSwitching</key>
	<true/>
</dict>
</plist>
```

Setting it to `true` maybe fix the issues. More info can be found here: https://developer.apple.com/library/archive/qa/qa1734/_index.html

**Comment by maxnet** [comment]
>Setting it to true maybe fix the issues.

I am not that familiar with Apple gear, and do not have a Macbook (only a simple Mac mini for development purposes)
So can you spell out what those issues are exactly?

Why is it a bad thing that our application -that indeed uses OpenGL graphics acceleration- uses the discrete GPU?
Is it just a power consumption thing, that would make it better if it could use the Intel GPU, or are there more serious issues?

**Comment by 3rror** [comment]
> Why is it a bad thing that our application -that indeed uses OpenGL graphics acceleration- uses the discrete GPU? Is it just a power consumption thing, that would make it better if it could use the Intel GPU, or are there more serious issues?

It's just to save battery life, and also reduce the heat produced resulting in less fan noise when combined with other intensive tasks.
Nothing _really serious_ in practice 😃. I noticed it because it happened to leave the tool open while writing multiple images.

I  have to check if I can build the Imager locally and see if the fix works. If I remember correctly, I have already applied it in the past to some Qt apps.
